deepspeed/runtime/fp16/fused_optimizer.py

```diff
@@ -261,7 +261,7 @@ def step(self, closure=None):
                 pg = self.deepspeed.mpu.get_data_parallel_group()
             else:
                 pg = groups._get_data_parallel_group()
-            all_groups_norm = get_norm_with_moe_layers(all_groups_norm, pg)
+            all_groups_norm = get_norm_with_moe_layers_fast(all_groups_norm, pg)
```

Comment:
Why not use the non-fast also for fp16? I think the "fast" version will give incorrect results when you are using multiple EP sizes in a single model (e.g. first layer uses EP=4 and 2nd layer uses EP=2).

Reply:
Yes, I have also unified the past fp16 with the current bf16.